src/coreComponents/mesh/generators/HexCellBlockManager.cpp

```diff
@@ -0,0 +1,1126 @@
+  {
+    localIndex f = m_uniqueFaces[curFace];
+    LocalCellIndex c = f / Hex::nbFacets;
+
+    // We want the nodes of face f % 6 in cell c 
+    auto blockCell = getBlockCellFromManagerCell(c) ;
+    localIndex blockIndex = blockCell.first;
+    localIndex cellInBlock = blockCell.second;
+
+    CellBlock const & cB = getCellBlock( blockIndex );
+    HexFacetIndex faceToStore = f % Hex::nbFacets;
+
+    assert(cellInBlock >= 0); // debug 
+
+    // Maybe the oriented facets would be useful - if they are the ones recomputed 
+    // later one by the FaceManager
+    faceToNodes[curFace][0] = cB.getElementNode(cellInBlock, Hex::facetVertex[faceToStore][0]);
+    faceToNodes[curFace][1] = cB.getElementNode(cellInBlock, Hex::facetVertex[faceToStore][1]);
+    faceToNodes[curFace][2] = cB.getElementNode(cellInBlock, Hex::facetVertex[faceToStore][2]);
+    faceToNodes[curFace][3] = cB.getElementNode(cellInBlock, Hex::facetVertex[faceToStore][3]);
+  }
+  return true;
+}
+
+
+bool HexMeshConnectivityBuilder::computeNodesToFaces( ArrayOfSets<localIndex> & nodeToFaces ) const
+{
+  ArrayOfArrays<localIndex> faceToNodes; 
+  computeFacesToNodes( faceToNodes );
+
+  // 1 - Counting  - Quite unecessary for hexahedral meshes 
+  //( 12 for regular nodes  - TODO check a good max for singular node )
+  std::vector<unsigned int> nbFacesPerNode(nbNodes, 0);
+  for(unsigned int i = 0; i < faceToNodes.size(); ++i)
+  {
+    nbFacesPerNode[faceToNodes[i][0]]++;
+    nbFacesPerNode[faceToNodes[i][1]]++;
+    nbFacesPerNode[faceToNodes[i][2]]++;
+    nbFacesPerNode[faceToNodes[i][3]]++;
+  }
+  localIndex valuesToReserve = std::accumulate(nbFacesPerNode.begin(), nbFacesPerNode.end(), 0);
+
+  // 2 - Allocating 
+  nodeToFaces.resize(0);
+  nodeToFaces.resize(nbNodes);
+  nodeToFaces.reserveValues( valuesToReserve );
+  for (localIndex n = 0; n < nbNodes; ++n)
+  {
+    nodeToFaces.appendSet( nbFacesPerNode[n] );
+  }
+  
+  // 3 - Filling 
+  for(unsigned int i = 0; i < faceToNodes.size(); ++i)
+  {
+    nodeToFaces.insertIntoSet( faceToNodes[i][0],i );
+    nodeToFaces.insertIntoSet( faceToNodes[i][1],i );
+    nodeToFaces.insertIntoSet( faceToNodes[i][2],i );
+    nodeToFaces.insertIntoSet( faceToNodes[i][3],i );
+  }
+
+  return false;
+}
+
+
+// TODO We have a problem - where are the Element index valid ?
+bool HexMeshConnectivityBuilder::computeFacesToElements( array2d<localIndex> & faceToElements ) const
+{
+  faceToElements.resize(0);
+  faceToElements.resize(nbFaces, 2);
+  // Initialize contents to NO_ID 
+  faceToElements.setValues< serialPolicy >( NO_ID );
+
+  for(unsigned int curFace = 0; curFace < m_uniqueFaces.size(); ++curFace)
+  {
+    localIndex f = m_uniqueFaces[curFace];
+    LocalFaceIndex neighbor = m_allFacesToNeighbors[f];
+
+    LocalCellIndex c = f/Hex::nbFacets;
+    LocalCellIndex cNeighbor = -1;
+    
+    localIndex cellInBlock = getBlockCellFromManagerCell(c).second;
+    localIndex neighborCellInMaybeNotSameBlock = -1;
```

Comment:
Despite the descriptive name, I quite do not get everything of what you mean with this variable.
At first, I thought it could be a `bool`.
Maybe change to a simpler name but improve comments?

Reply:
The question is: where are (and should be)  the Element indices valid in the mappings  FaceToElements / NodeToElements / EdgeToElements?